OSDOCS-10287 OCP 4.15.10 RNs (#75108)
openshift/openshift-docs: release_notes/ocp-4-15-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

[id="ocp-4-15-10"]
=== RHSA-2024:1887 - {product-title} 4.15.10 bug fix and security update

Issued: 2024-04-25

> {product-title} release 4.15.10, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:1887[RHSA-2024:1887] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:1892[RHSA-2024:1892] advisory.

Reviewer: link:https://access.redhat.com/errata/RHSA-2024:1892[RHSA-2024:1892] not currently working, but I do realize it just may not be live yet....

Author: It's not live yet. Thanks for checking!

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.15.10 --pullspecs
----

[id="ocp-4-15-10-bug-fixes"]
==== Bug fixes

> * Previously, Clusters born before 4.7 will had signer keys for api-int endpoint updated unexpectedly when users upgraded to OpenShift Container Platform 4.15 due to the installer deleting the SecretTypeTLS and then recreating the secret with `kubernetes.io/tls` type.  With this release, the issue is resloved by the installer changing the secret type without deleting the secret. (link:https://issues.redhat.com/browse/OCPBUGS-31807[*OCPBUGS-31807*])

Reviewer: ```suggestion
* Previously, Clusters created before 4.7 will have had signer keys for api-int endpoint updated unexpectedly when users upgraded to OpenShift Container Platform 4.15 due to the installer deleting the SecretTypeTLS and then recreating the secret with `kubernetes.io/tls` type.  With this release, the issue is resolved by the installer changing the secret type without deleting the secret. (link:https://issues.redhat.com/browse/OCPBUGS-31807[*OCPBUGS-31807*])
```

> * Previously, Clusters created before OpenShift Container Platform 4.7 will have had signer keys for api-int endpoint updated unexpectedly when users upgraded to OpenShift Container Platform 4.15 due to the installer deleting the SecretTypeTLS and then recreating the secret with `kubernetes.io/tls` type.  With this release, the issue is resolved by the installer changing the secret type without deleting the secret. (link:https://issues.redhat.com/browse/OCPBUGS-31807[*OCPBUGS-31807*])

Reviewer 2: ```suggestion
* Previously, clusters created before {product-title} 4.7 will have had signer keys for api-int endpoint updated unexpectedly when users upgraded to {product-title} 4.15 due to the installer deleting the SecretTypeTLS and then recreating the secret with `kubernetes.io/tls` type.  With this release, the issue is resolved by the installer changing the secret type without deleting the secret. (link:https://issues.redhat.com/browse/OCPBUGS-31807[*OCPBUGS-31807*])
```

> * Previously,  when users imported image stream tags, ICSP (In-Circuit Serial Programming) and IDMS (Integrated Database Managing System) were not allowed to co-exist. OpenShift ignored any IDMS/ITMS created by the user and favored ICSP. With this release, they are allowed to co-exist so importing image stream tags will now respect IDMS/ITMS when ICSP is also present. (link:https://issues.redhat.com/browse/OCPBUGS-31469[*OCPBUGS-31469*])

Reviewer: ```suggestion
* Previously,  when users imported image stream tags, ICSP (In-Circuit Serial Programming) was not allowed to co-exist with IDMS (Integrated Database Managing System) and ImageTagMirrorSet (ITMS). OpenShift ignored any IDMS/ITMS created by the user and favored ICSP. With this release, they are allowed to co-exist as importing image stream tags will now respect IDMS/ITMS when ICSP is also present. (link:https://issues.redhat.com/browse/OCPBUGS-31469[*OCPBUGS-31469*])
```

Reviewer: Not sure if this is the right way to go about it (my suggestion), but ITMS is not addressed specifically. Not sure if it needs to be?

> * Previously, when users imported image stream tags, ICSP (In-Circuit Serial Programming) was not allowed to co-exist with IDMS (Integrated Database Managing System) and ImageTagMirrorSet (ITMS). OpenShift ignored any IDMS/ITMS created by the user and favored ICSP. With this release, they are allowed to co-exist as importing image stream tags will now respect IDMS/ITMS when ICSP is also present. (link:https://issues.redhat.com/browse/OCPBUGS-31469[*OCPBUGS-31469*])

Reviewer 2: ICSP (In-Circuit Serial Programming)
IDMS (Integrated Database Managing System)
ImageTagMirrorSet (ITMS)

Suggesting parallelism, unless there's a reason you didn't want to: 

```suggestion
* Previously, when users imported image stream tags, ICSP (In-Circuit Serial Programming) was not allowed to co-exist with IDMS (Integrated Database Managing System) and  ITMS (ImageTagMirrorSet). OpenShift ignored any IDMS/ITMS created by the user and favored ICSP. With this release, they are allowed to co-exist as importing image stream tags will now respect IDMS/ITMS when ICSP is also present. (link:https://issues.redhat.com/browse/OCPBUGS-31469[*OCPBUGS-31469*])
```

> * Previously, Terraform would create the compute server group with the policy set for the control plane. As a consequence, the 'serverGroupPolicy' property of `install-config.yaml` was ignored for the Computes. With this release, the server group policy set in `install-config.yaml` for the compute MachinePool is correctly applied at install-time in the Terraform flow.  (link:https://issues.redhat.com/browse/OCPBUGS-31335[*OCPBUGS-31335*])

Reviewer: Should "Computes" be capitalized?

> * Previously, Terraform would create the compute server group with the policy set for the control plane. As a consequence, the 'serverGroupPolicy' property of `install-config.yaml` was ignored for the compute server group. With this release, the server group policy set in `install-config.yaml` for the compute MachinePool is correctly applied at install-time in the Terraform flow.  (link:https://issues.redhat.com/browse/OCPBUGS-31335[*OCPBUGS-31335*])

Reviewer 2: I think generally we're supposed to follow `.yaml` with a noun:

```suggestion
* Previously, Terraform would create the compute server group with the policy set for the control plane. As a consequence, the 'serverGroupPolicy' property of the `install-config.yaml` file was ignored for the compute server group. With this release, the server group policy set in the `install-config.yaml` file for the compute MachinePool is correctly applied at install-time in the Terraform flow.  (link:https://issues.redhat.com/browse/OCPBUGS-31335[*OCPBUGS-31335*])
```

> * Previously, Projects that specified a non-intersecting openshift.io/node-selector project selector with pods `.spec.nodeName` could cause runaway Pod creation in Deployments. With this release, pods with non-intersecting `.spec.nodeName` will not be admitted by the API server which resolves the issue. (link:https://issues.redhat.com/browse/OCPBUGS-29922[*OCPBUGS-29922*])

Reviewer 2: ```suggestion
* Previously, projects that specified a non-intersecting openshift.io/node-selector project selector with pods `.spec.nodeName` could cause runaway Pod creation in Deployments. With this release, pods with non-intersecting `.spec.nodeName` are not admitted by the API server which resolves the issue. (link:https://issues.redhat.com/browse/OCPBUGS-29922[*OCPBUGS-29922*])
```

> * Previously, a remote attacker with basic login credentials could check the pod manigest to discover a repository pull secret. With this release, the vulnerability has been fixed. (link:https://issues.redhat.com/browse/OCPBUGS-28769[*OCPBUGS-28769*])

Reviewer: ```suggestion
* Previously, a remote attacker with basic login credentials could check the pod manifest to discover a repository pull secret. With this release, the vulnerability has been fixed. (link:https://issues.redhat.com/browse/OCPBUGS-28769[*OCPBUGS-28769*])
```
